include/deal.II/lac/vector_operations_internal.h

```diff
@@ -1248,11 +1252,13 @@ namespace internal
      */
     template <typename Operation, typename ResultType>
     void parallel_reduce (const Operation   &op,
-                          const size_type    vec_size,
+                          const size_type    end,
                           ResultType        &result,
-                          std_cxx11::shared_ptr<parallel::internal::TBBPartitioner> &partitioner)
+                          std_cxx11::shared_ptr<parallel::internal::TBBPartitioner> &partitioner,
+                          const size_type start = 0)
```

Comment:
But adjusting all callers is pretty mechanical and should be straight-forward and safes problems down the road.